File: 953-reverse-only-letters/reverse-only-letters.py
class Solution(object):
    def reverseOnlyLetters(self, s):
        """
        :type s: str
        :rtype: str
        """
        left=0
        right=len(s)-1
        arr=list(s)
        while(left<right):
            while(left<right and not arr[left].isalpha()): left+=1
            while(left<right and not arr[right].isalpha()): right-=1
            arr[left],arr[right]=arr[right],arr[left]
            left+=1
            right-=1
        return "".join(arr)

File: 953-reverse-only-letters/reverse-only-letters.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function(s) {
    let left=0;
    let right=s.length-1;
    let arr=s.split("");
    while(left<right){
        while(left<right && !/[a-zA-Z]/.test(arr[left])) left++;
        while(left<right && !/[a-zA-Z]/.test(arr[right])) right--;
        [arr[left],arr[right]]=[arr[right],arr[left]];
        left++;
        right--;
    }
    return arr.join("");
};